### PR TITLE
FEAT: Kick up unauthorized_client error

### PR DESCRIPTION
**Pull request by mindofbeholder**
This will catch the "unauthorized_client" error that is seen when someone sets up their reddit application as a "web app" and attempts to authenticate with a username & password instead of a token.

closes not-an-aardvark/snoowrap#93

### CODE REVIEW

**Comment by mindofbeholder** [comment]
Or we could just directly kick up the actual error message. I was just looking to make the message a touch more user friendly.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the pull request.

Would it work to add a check for any `error` property on a response object, rather than ? That way, errors can be handled in general without needing to hardcode specific error messages.

File: src/request_handler.js [hunk 1]
```diff
@@ -208,6 +208,10 @@ export function updateAccessToken () {
       this.tokenExpiration = Date.now() + (tokenInfo.expires_in * 1000);
       if (tokenInfo.error === 'invalid_grant') {
         throw new Error('"Invalid grant" error returned from reddit. (You might have incorrect credentials.)');
+      } else if (tokenInfo.error_description !== undefined) {
+        throw new Error(`Reddit return an error. ${tokenInfo.error}: ${tokenInfo.error_description}`);
+      } else if (tokenInfo.error !== undefined) {
+        throw new Error(`Reddit return an error: ${tokenInfo.error}`);
```
Comment (not-an-aardvark) [inline src/request_handler.js]: Nitpick: I think these should say "Reddit returned an error" (with "ed" at the end of "return").

**Comment by mindofbeholder** [comment]
I wouldn't call it nitpicking. More like having standards. Not sure how I managed to phrase it in that way. Sorry about that.

**Pull request merged**